proxy/proxy.go

```diff
@@ -139,3 +164,16 @@ func (proxy *Proxy) weaveCIDRsFromConfig(config *docker.Config, hostConfig *dock
 	}
 	return nil, nil
 }
+
+func (proxy *Proxy) addWeaveWaitVolume(hostConfig *docker.HostConfig) {
+	var binds []string
+	for _, bind := range hostConfig.Binds {
+		s := strings.Split(bind, ":")
+		if len(s) >= 2 && s[1] == "/w" {
+			continue
+		}
+		binds = append(binds, bind)
+	}
+	hostConfig.Binds = binds
+	hostConfig.Binds = append(hostConfig.Binds, fmt.Sprintf("%s:/w:ro", proxy.weaveWaitVolume))
```

Comment:
ditch the `hostConfig.Binds = binds` and simply write "append(binds, ...)" in the next line.